server/src/main/java/com/google/fhir/proxy/HttpFhirClient.java

```diff
@@ -43,7 +43,38 @@ public abstract class HttpFhirClient {
 
   // The list of header names to keep in a response sent from the proxy; use lower case only.
   // Note we don't copy content-length/type because we may modify the response.
-  static final Set<String> HEADERS_TO_KEEP = Sets.newHashSet("last-modified", "date");
+  static final Set<String> RESPONSE_HEADERS_TO_KEEP =
```

Comment:
Please consider adding a reference to the FHIR documentation from which these HEADERs come from (for both sets). I think something like what you have in the PR description is good enough (doesn't need to be detailed for each).